usb_ser_mon/usb_ser_mon.py

```diff
@@ -178,38 +183,38 @@ def usb_serial_mon(monitor, device, baud=115200, debug=False, echo=False):
                     serial_port.close()
                     return
                 if debug:
-                    for x in data:
-                        log.print("Serial.Read '%c' 0x%02x\r" % (x, ord(x)))
+                    for x in six.iterbytes(data):
+                        log.print("Serial.Read 0x%02x\r" % x)
                 pos = 0
                 while True:
-                    nl_pos = data.find('\n', pos)
+                    nl_pos = data.find(b'\n', pos)
                     if nl_pos < 0:
                         break
-                    if nl_pos > 0 and data[nl_pos - 1] == '\r':
+                    if nl_pos > 0 and data[nl_pos-1:nl_pos] == b'\r':
                         # already have \r before \n - just leave things be
                         pos = nl_pos + 1
                         continue
-                    data = data[:nl_pos] + '\r' + data[nl_pos:]
+                    data = data[:nl_pos] + b'\r' + data[nl_pos:]
                     pos = nl_pos + 2
                 sys.stdout.write(data)
                 sys.stdout.flush()
-                log.log(data, end='')
+                log.log(data)
             if fileno == sys.stdin.fileno():
                 data = sys.stdin.read(1)
                 if len(data) == 0:
                     continue
                 if debug:
-                    for x in data:
-                        log.print("stdin.Read '%c' 0x%02x\r" % (x, ord(x)))
-                if data[0] == EXIT_CHAR:
+                    for x in six.iterbytes(data):
+                        log.print("stdin.Read 0x%02x\r" % x)
```

Comment:
Same thing here. I'd prefer to see the original where if logs the character and the hex value.

If I was going to change anything, I'd write a helper function which takes an int and returns ord('?') for non-ascii characters (i.e. < 0x20 or > 0x7e)